### DISCUSSION TITLE
Bad GPX recording

### DISCUSSION THREAD

**Discussion by jimackrel** [osmandapp/OsmAnd#17930] [Q&A]
Hello
I use gpx recording when I bicycle and have done so reliably for a couple of years. Since the 8/1/23 update you are writing the gpx file differently, about 1/10 the size for the same route. The application still draws the older gpx files correctly, i.e. not riding through lakes and canals, but the recent gpx files are missing a lot of data. I have attached a gpx from 7/30/23 and a subsequent one from 8/04/23. You’ll see large differences in what is written in each packet. My wife uses the same application and sees the same problem. Both phones are Samsung S22+, OneUI v5.1, Android v13, playsystem update 6/1/2023, security patch level 7/1/2023. Osmand+ 4.5.9 released 8/1/2023. Previous version was Osmand+ 4.4.7. The only thing that has changed coincidentally with the problem’s appearance is your version. I have never used anything other than 5 sec interval for recording. I can reproduce the problem by using it as described. It is not intermittent. I raised this issue via email, I was sent here.

[OSMA-gpx.zip](https://github.com/osmandapp/OsmAnd/files/12352706/OSMA-gpx.zip)

**Comment by DmitryAlexei** [comment]
Second track lacks some data. This can be GPS signal loss problem due to Android system settings or wrong configuration of Trip recording plugin. Please:
- reset to default trip recording plugin (https://docs.osmand.net/docs/user/plugins/trip-recording/#recording-settings)
- remove OsmAnd from *deep sleep* (https://www.samsung.com/us/support/answer/ANS00088422/) if it was accidentally put in there
- you can also add OsmAnd to the list of never sleeping apps
- make sure that Google services are set as the Location source (you can also try Android API and see if it is better)

**Comment by jimackrel** [comment]
Hello

I reset the plugin settings as below, checked that it wasn’t deep sleeping (it wasn’t) set for never sleeping app, no difference to the output. As I stated, this is a sudden change coincident with your version change. I don’t think there were any system updates in the same time period. I have 400+ files that are OK, but everything since 8/4/23 is bad. Can I revert to the previous version to verify that it’s not a Samsung/Android issue? FWIW, Google maps timeline seems correct.

 

Jim

**Reply by pebogufi** [reply]
Here you can download older versions, but it is NOT shure that they can be installed if they are too old:
https://download.osmand.net/releases/
Just try it.

**Reply by DmitryAlexei** [reply]
Don't forget to export important data https://docs.osmand.net/docs/user/navigation/setup/route-details#share--export-actions

**Comment by jimackrel** [comment]
Is this the + version? Do I need to uninstall existing first?

**Comment by Lucas-Sichardt** [comment]
Hello,

I just experience the same problem I think.
 The new gpx file is really small and other applications do not open it. Marble for instance tells that there are syntax errors and open remaining brackets.

Is there perhaps some issue with the "end" action finalizing the file?
I had a good track on the display while osmand recorded and the screen was always on as it was a test ride if about 2,5km - and the sane for the second test in the opposite direction. Both files cannot be opened in other applications...

**Comment by jimackrel** [comment]
I switched to android app as location source and it has worked OK since.
The problem I saw was very intermittent capture, so maybe 3 or so times a
minute instead of every 5s. I think the track was obviously bad (cutting
corners etc) but I didn't look at it until I stopped.
